### ISSUE TITLE
[A11y] Search Results - Grid - "SAVE SELECTED TO" button/modal

### ISSUE THREAD

**Issue by jechols** [OregonDigital/OD2#1931]
### Operating System

Windows

### Browser

Edge

### Affected page

Search

### Admin user?

Yes

### Issue Description

This has some similarities to #1840, but this particular modal has extra complications.

- Tab loses focus for a moment when you're at the last focusable element - you have to tab twice to get to the first element again. This is extra confusing because focus seems to truly disappear. Nothing is announced in NVDA.
- Shift-tab gets "stuck" instead of wrapping around to the last element.
- The close button seems to be blocked from assistive tech somehow. I don't understand it. I can only activate it when NVDA is turned off. After some fiddling, it turns out this is due to there being an `aria-hidden="true"` attribute on the button.
  - If you remove the `aria-hidden` attribute, you end up with "times" as the button's accessible name.

Within the modal, there are other issues:

- The accessible text for the listbox / fieldset / legend area is confusing. If I tab forward to get to the dropdown, the group is announced as `MY COLLECTIONS grouping invalid entry Or,`. When I shift-tab and go backward to the save button (the last item in the grouping), it's announced correctly: `MY COLLECTIONS  grouping`. I don't know if this is an NVDA oddity because that area is coded really strangely. The fieldset, for instance, is outside the form.
- "Or, " should be within the button or perhaps not there at all. Right now the word "Or" is really only there for sighted users, and it sort of implies a flow or connection which either everybody needs to know about, or maybe nobody.
- The "dropdown" is a bit weird on its own. The AJAXy bits seem like they work in terms of accessibility, but it's hard to test - I can't find any collections. But the accessible name is really hard to pin down because of how the widget is coded; I think it's just a div that only gets behaviors added when it gets focused or something. NVDA announces this as "Select a collection... menubutton has autocomplete editable blank". "menubutton" is definitely not the right role for this widget.
- "SAVE THIS ITEM" functions as a heading but isn't coded as one

### Suggested Fix

The modal behavior:

- There should never be a "blank" focus when tabbing forward or backward.  Forward tabbing past the end should always take you to the first focusable element within the dialog.  The reverse should also be true.
- Focusable elements must never have `aria-hidden="true"`. Un-hide this and give it a useful label, e.g., "Close".
- Reference: https://www.w3.org/TR/wai-aria-practices-1.1/#dialog_modal

Within the modal:

- The easiest way to fix the odd "MY COLLECTIONS group invalid entry Or," is to get that fieldset out of there entirely. It doesn't really make sense semantically to use a fieldset to group a dropdown and a button, nor to have a fieldset around the entire form.
- The "Or, " text could be removed in my opinion and instead a simple explanation could be put above the form. The way it's laid out right now, it's going to be a challenge for many users to understand. Screen reader users will struggle the most, but also users with poor cognition.
- The "Select a collection" dropdown should probably be built as a combo box, but these are notoriously difficult to make fully accessible. Some information: https://www.w3.org/TR/wai-aria-practices-1.1/#combobox
  - It may be easier to look for a different approach to this problem, though I'm not sure what approach that might be....
- "SAVE THIS ITEM" should probably be a level two heading

### Can anybody QA or does this need an expert?

Needs expert

### QA Steps

This really needs screen reader use and HTML inspection to validate.  If it were possible, this would be a place to get professional testing services from real users of assistive tech.

**Comment by jechols** [comment]
I still can't get the dialog to show collections - I think the a11y issues are dealt with, but that bit will need testing when it's fixed, I think
